hw16/Team.cpp

```diff
@@ -0,0 +1,36 @@
+#include "Team.h"
+#include "Player.h"
+#include <iostream>
+
+Team::Team(const std::string& name) : m_name(name), m_numPlayers(0) {}
+
+bool Team::addPlayer(Player* player) {
+    if (m_numPlayers < MAX_PLAYERS) {
+        m_Players[m_numPlayers++] = player;
+        player->setTeam(this);
+        return true;
+    }
+    return false;
+}
+
+bool Team::removePlayer(Player* player) {
+    for (int i = 0; i < m_numPlayers; i++) {
+        if (m_Players[i] == player) {
+            for (int j = i; j < m_numPlayers - 1; j++) {
+                m_Players[j] = m_Players[j + 1];
+            }
+            m_numPlayers--;
+            player->setTeam(nullptr);
+            return true;
+        }
+    }
+    return false;
+}
+
+void Team::getInfo(bool isFullInfo) const {
```

Comment:
I usually prefer going with 2 methods in similar cases. Like 
getFullTeamInfo() / getTeamInfoWithPlayers() ...
getTeaminfo() / getTeamName()...
But its more like a style